src/SimplePie.php

```diff
@@ -2027,10 +2031,47 @@ protected function fetch_data(&$cache)
                     }
                     // Now also do feed discovery, but if microformats were found don't
                     // overwrite the current value of file.
-                    $discovered = $locate->find(
-                        $this->autodiscovery,
-                        $this->all_discovered_feeds
+                    $possible_feed_urls = $detector->discover_possible_feed_urls(
+                        $file,
+                        $this->autodiscovery
                     );
+
+                    $discovered = null;
+                    $checked_feeds = 0;
+
+                    foreach ($possible_feed_urls as $href) {
```

Comment:
I like having a pure (side-effect-free detector) but moving the effects to `SimplePie` class is not nice when it is already overly bloated. Could we have a separate helper class that would encapsulate this code?